deepchem/models/torch_models/chemberta.py

```diff
@@ -0,0 +1,174 @@
+
+class ChemBERTaMaskedLMModel(TorchModel):
+    def __init__(self,
+                vocab_size: int = 600,
+                max_position_embeddings = 515,
+                number_attention_heads=12,
+                num_hidden_layers=6,
+                dataset_path: str = '',
+                mode='pre-trained',
+                model_path='DeepChem/SmilesTokenizer_PubChem_1M',
+                mlm_probability: float = 0.15,
+                output_dir = 'model/',
+                frac_train: int = 0.95,
+                eval_steps: int = 1000,
+                logging_steps: int = 100,
+                overwrite_output_dir: boolean = True,
+                num_train_epochs: int = 10,
+                per_device_train_batch_size: int = 64,
+                save_steps :int = 10000,
+                save_total_limit: int = 2,
+                fp16: bool= True,
+                run_name: str = 'chemberta_lm'
+                **kwargs):
+        
+        from transformers import LineByLineTextDataset
+        from transformers import DataCollatorForLanguageModeling
+        from transformers import TrainingArguments
+
+        model = ChemBERTa(
+            vocab_size=vocab_size,
+            max_position_embeddings = max_position_embeddings,
+            number_attention_heads = number_attention_heads,
+            num_hidden_layers = num_hidden_layers,
+            dataset_path = dataset_path,
+            mode = mode,
+            model_path = model_path,
+            model_output_dir = model_output_dir,
+            **kwargs)
+                
+        dataset = LineByLineTextDataset(file_path=dataset_path, 
+                                            block_size=512)
+
+        train_size = max(int(frac_train * len(dataset)), 1)
+        eval_size = len(dataset) - train_size
+        print(f"Train size: {train_size}")
+        print(f"Eval size: {eval_size}")
+
+        self.train_dataset, self.eval_dataset = random_split(dataset, [train_size, eval_size])
+
+        self.data_collator = DataCollatorForLanguageModeling(
+                            tokenizer=model.tokenizer, mlm=True, 
+                            mlm_probability=mlm_probability)
+
+        is_gpu = torch.cuda.is_available()
+        
+        self.training_args = TrainingArguments(
+            evaluation_strategy="steps",
+            eval_steps=eval_steps,
+            load_best_model_at_end=True,
+            logging_steps=logging_steps,
+            output_dir=os.path.join(output_dir, FLAGS.run_name),
+            overwrite_output_dir=overwrite_output_dir,
+            num_train_epochs=num_train_epochs,
+            per_device_train_batch_size=per_device_train_batch_size,
+            save_steps=save_steps,
+            save_total_limit=save_total_limit,
+            fp16 = is_gpu and fp16, # fp16 only works on CUDA devices
+            report_to="wandb",
+            run_name=run_name,
+        )
+    
+    def _fit(self):
```

Comment:
@rbharath Would love your thoughts on if it is ok to overwrite the TorchModel fit so we can fit the model into the Trainer object instead

Reply:
It's ok to overwrite if necessary but not if we can't satisfy the standard TorchModel API. I think write now the user would be surprised since they can't train this model with DeepChem datasets or standard deepchem flows